### PR TITLE
JLA-12

### PR DESCRIPTION
**Pull request by Dawid55a**

### CODE REVIEW

File: API/module/manga/manga_repository.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    async def find_all(self) -> List[Manga]:
+        session: AsyncSession
+        async with self.session_maker() as session:
+            return (await session.execute(select(Manga))).all()
+
+    async def insert(self, name: str, img_path: str, extension: str) -> None:
+        manga = Manga(name=name, img_path=img_path, extension=extension)
+        async with self.session_maker() as session:
+            session.add(manga)
+            await session.commit()
```
Comment (DamianWDev) [inline API/module/manga/manga_repository.py:22]: [find_all] has defined session type but [insert] doesn't?

File: API/module/manga/manga_route.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+from API.module.manga.manga_service import MangaService
+from API.container import Container
+from API.module.shared.file_manager import UnsupportedExtensionException
+
+router = APIRouter(
+    prefix='/manga'
+)
+
+
+@router.get('/mangas/', status_code=status.HTTP_201_CREATED, response_class=ORJSONResponse)
```
Comment (DamianWDev) [inline API/module/manga/manga_route.py:14]: created status in get EP?

**Review by DamianWDev — CHANGES_REQUESTED** [review]

File: API/module/db/model/manga.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    id = Column(Integer, primary_key=True)
+    name = Column(String(255))
+    img_path = Column(String(255))
+    format = Column(String(4))
+
+    def __repr__(self):
+        return f"""
+        Manga(id={self.id}, 
+        name={self.name}, 
+        img_path={self.img_path})
```
Comment (DamianWDev) [inline API/module/db/model/manga.py:18]: missing img_extension

File: API/module/manga/manga_repository.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    def __init__(self, session_maker: sessionmaker):
+        self.session_maker = session_maker
+
+    async def find_all(self) -> List[Manga]:
+        session: AsyncSession
+        async with self.session_maker() as session:
+            return (await session.execute(select(Manga))).all()
+
+    async def insert(self, name: str, img_path: str, extension: str) -> None:
+        manga = Manga(name=name, img_path=img_path, extension=extension)
```
Comment (DamianWDev) [inline API/module/manga/manga_repository.py:19]: rename img_extension

File: API/module/manga/manga_service.py [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    def __init__(self, file_manager: FileManager, manga_repository: MangaRepository) -> None:
+        self.file_manager = file_manager
+        self.manga_repository = manga_repository
+
+    async def get_mangas(self):
+        return await self.manga_repository.find_all()
+
+    async def add_manga(self, file: UploadFile, name: str) -> None:
+        filename, extension = await self.file_manager.save(file)
+        await self.manga_repository.insert(name, filename, extension)
```
Comment (DamianWDev) [inline API/module/manga/manga_service.py:17]: rename img_extension

**Review by DamianWDev — CHANGES_REQUESTED** [review]
names, useless whitespaces lines in imports, names, can't run app, NAMES!

File: API/module/db/database.py [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+class Database:
+
+    def __init__(self, db_url: str = DB_STRING) -> None:
+        self._engine = create_async_engine(db_url, future=True, echo=True)
+        self.session_maker = sessionmaker(self._engine, expire_on_commit=False, class_=AsyncSession)
+
+    async def create_database(self) -> None:
+        async with self._engine.begin() as conn:
+            await conn.run_sync(Base.metadata.drop_all)
+            await conn.run_sync(Base.metadata.create_all)
```
[the remark this answers is not in the document]
Comment (Dawid55a) [reply API/module/db/database.py:20]: Will be removed when miggration will be added

**Pull request merged**